Sources_v3/Controllers/ChannelListController.swift

```diff
@@ -179,8 +195,8 @@ extension ChannelListControllerGeneric where ExtraData == DefaultDataTypes {
     /// limits of Swift and the way it handles protocols with associated types, it's required to use `setDelegate` method
     /// instead to set the delegate, if you're using custom extra data types.
     public weak var delegate: ChannelListControllerDelegate? {
-        set { anyDelegate = AnyChannelListControllerDelegate(newValue) }
-        get { anyDelegate?.wrappedDelegate as? ChannelListControllerDelegate }
+        set { multicastDelegate.mainDelegate = AnyChannelListControllerDelegate(newValue) }
+        get { multicastDelegate.mainDelegate?.wrappedDelegate as? ChannelListControllerDelegate }
```

Comment:
I think you can keep `delegate` as is and use `didSet`. The benefit of that you can remove `mainDelegate` from `MulticastDelegate`.

Reply:
> I think you can keep delegate as is and use didSet. 

What `didSet`? There's no did set in this variable.

> The benefit of that you can remove mainDelegate from MulticastDelegate.

I don't want to remove `mainDelegate`. It's the public-facing delegate. I want to be sure the following is valid:
```swift
controller.delegate = self
assert(controller.delegate === self)
```

Reply:
I thought about this:
```swift
public weak var delegate: ChannelListControllerDelegate? {
    didSet {
       multicastDelegate.addDelegate(delegate)
    }
}
```

Reply:
The `delegate` variable lives in the extension and is accessible only when `ExtraData == DefaultExtraData`. It can't be a stored property.